src/cgutils.cpp

```diff
@@ -1006,20 +1067,21 @@ static std::pair<Value*, bool> emit_isa(const jl_cgval_t &x, jl_value_t *type, c
 
     // intersection with Type needs to be handled specially
     if (jl_has_intersect_type_not_kind(type)) {
-        Value *vx = boxed(x, ctx);
+        Value *vx = mark_callee_rooted(boxed(x, ctx));
```

Comment:
I included not-a-safepoint in the definition of callee rooted for now because it's the same for all practical purposes. In a follow on commit, I plan to make that notion explicit. At that point, I'll clean this up.